SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialRPCService.cpp

```diff
@@ -225,7 +243,18 @@ TArray<SpatialRPCService::UpdateToSend> SpatialRPCService::GetRPCsAndAcksToSend(
 		SpatialRPCService::UpdateToSend& UpdateToSend = UpdatesToSend.AddZeroed_GetRef();
 		UpdateToSend.EntityId = It.Key.EntityId;
 		UpdateToSend.Update.component_id = It.Key.ComponentId;
-		UpdateToSend.Update.schema_type = It.Value;
+		UpdateToSend.Update.schema_type = It.Value.Update;
+		if (It.Value.SpanIds.Num() > 1)
+		{
+			// Attach causes together, we need to send these span ids with the RPCs to continue on the other side without
+			// joining unrelated RPCs together.
+			UpdateToSend.SpanId = Trace_EventTracer_AddSpan(EventTracer->GetWorkerEventTracer(), &It.Value.SpanIds[0], It.Value.SpanIds.Num());
+		}
+		else if(It.Value.SpanIds.Num() == 1) 
+		{
+			// No need to chain causes here
+			UpdateToSend.SpanId = It.Value.SpanIds[0];
```

Comment:
This is one of the conventions I mentioned on the call. I'd advise against special casing circumstances where you would only end up specifying one cause for a span for the sake of consistency in the event tracing data (otherwise you may have event A and event B attached to the same span in this case but split across different spans in the case of multiple causes).